Command/ResolveCacheCommand.php

```diff
@@ -110,11 +129,23 @@ private function runCacheImageResolve(string $image, string $filter, bool $force
         $this->io->group($image, $filter, 'blue');
 
         try {
-            if ($forced || !$this->cacheManager->isStored($image, $filter)) {
-                $this->cacheManager->store($this->filterManager->applyFilter($this->dataManager->find($filter, $image), $filter), $image, $filter);
-                $this->io->status('resolved', 'green');
-            } else {
-                $this->io->status('cached', 'white');
+            $basePathContainer = new FilterPathContainer($image);
+            $filterPathContainers = [$basePathContainer];
+
+            if ($this->webpGenerate) {
+                $filterPathContainers[] = $basePathContainer->createWebp($this->webpOptions);
+            }
+
+            foreach ($filterPathContainers as $filterPathContainer) {
+                if ($forced || !$this->cacheManager->isStored($filterPathContainer->getTarget(), $filter)) {
+                    $binary = $this->dataManager->find($filter, $filterPathContainer->getSource());
+                    $filteredBinary = $this->filterManager->applyFilter($binary, $filter, $filterPathContainer->getOptions());
+                    $this->cacheManager->store($filteredBinary, $filterPathContainer->getTarget(), $filter);
+
+                    $this->io->status('resolved', 'green');
+                } else {
+                    $this->io->status('cached', 'white');
+                }
             }
 
             $this->io->line(sprintf(' %s', $this->cacheManager->resolve($image, $filter)));
```

Comment:
Interesting idea :+1:

Reply:
Of course, it would be nice to see URLs to WebP pictures in the console, but this requires opening the internal implementation of `FilterService::buildFilterPathContainers()` to the outside. I wouldn't want that.